src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+            if polygon_mesh.is_volume:
+                if tol_size is not None and tol_size >= 0.01:
+                    warnings.warn(f"Computing bounded footprint of polygon resulted in invalid volume, "
+                                  f"but was able to remedy this by buffering/simplifying polygon by {tol_size}")
+                break
+
+        if not polygon_mesh.is_volume:
+            raise RuntimeError("Computing bounded footprint of polygon resulted in invalid volume")
+
+        return MeshVolumeRegion(polygon_mesh, centerMesh=False)
+
+    def buffer(self, amount):
+        buffered_polygon = self.polygon.buffer(amount)
+
+        if isinstance(buffered_polygon, shapely.geometry.Polygon):
+            buffered_polygon = MultiPolygon([buffered_polygon])
+
+        return PolygonalFootprintRegion(polygon=buffered_polygon, name=self.name)
+
+    @cached_property
+    def isConvex(self):
+        return self.polygons.equals(self.polygons.convex_hull)
+
+class PathRegion(Region):
+    """A region composed of multiple polylines in 3D space.
+
+    One of points or polylines should be provided. If both are provided,
+    points will be used.
+
+    Args:
+        points: A list of points defining a single polyline.
+        polylines: A list of list of points, defining multiple polylines.
+    """
+    def __init__(self, points=None, polylines=None, name=None):
+        super().__init__(name)
+        # Standardize inputs
+        if points is not None:
+            polylines = [points]
+        elif polylines is not None:
+            pass
+        else:
+            raise ValueError("PathRegion created without specifying points or polylines")
+
+        # Extract a list of vertices and edges, without duplicates
+        self.vec_to_vert = {}
+        self.edges = []
+        vertex_iter = 0
+
+        for pls in polylines:
+            last_pt = None
+
+            for pt in pls:
+                # Extract vertex
+                cast_pt = toVector(pt)
+
+                if cast_pt not in self.vec_to_vert:
+                    self.vec_to_vert[cast_pt] = vertex_iter
+                    vertex_iter += 1
+
+                # Extract edge, if not first point
+                if last_pt is not None:
+                    new_edge = self.vec_to_vert[last_pt], self.vec_to_vert[cast_pt]
+                    self.edges.append(new_edge)
+
+                # Update last point
+                last_pt = cast_pt
+
+        self.vert_to_vec = {val:key for key,val in self.vec_to_vert.items()}
+        self.vertices = list(sorted(self.vec_to_vert.keys(), key=lambda vec: self.vec_to_vert[vec]))
+
+        # Extract length of each edge
+        self.edge_lengths = []
+
+        for edge in self.edges:
+            v1, v2 = edge
+            c1, c2 = self.vert_to_vec[v1], self.vert_to_vec[v2]
+
+            self.edge_lengths.append(c1.distanceTo(c2))
+
+    @lru_cache(maxsize=None)
+    def containsPoint(self, point, epsilon=1e-8):
```

Comment:
It's dubious to change the signature of `containsPoint`: this should really be a tolerance set on the region. Also the meaning of `epsilon` is unclear: using the distance from the edge would be more consistent with the other tolerances.

I'm not sure it makes sense to memoize this function either: we don't do it for other `containsPoint` methods which could be just as expensive. Certainly if we do memoize it we should impose a size limit on the cache, since it should not grow without bound over many simulations (as could easily happen for regions representing centerlines of roads, for example).